mystery-deckbuilder/Assets/Scripts/Encounter/EncounterScript.cs

```diff
@@ -7,31 +8,97 @@
  * script for the encounter gameobject
  * to use:
  *      encounter = Instantiate(encounter);
-        encounter.GetComponent<EncounterScript>().StartEncounter(int complianceThreshold, int startingCompliance, int startingPatience)
+        encounter.GetComponent<EncounterScript>().StartEncounter()
  */
 public class EncounterScript : MonoBehaviour
 {
     [SerializeField] private int _threshold;
     [SerializeField] private int _compliance;
     [SerializeField] private int _patience;
+
+    [SerializeField] private string _weakness; //using strings because that is how element is stored in the card class
+    [SerializeField] private string _resistance; // !!!CARD CLASS AND NPC CLASS STRINGS MUST MATCH!!! 
+
+    [SerializeField] private int _handSize = 5;
+    private int playCount = 0;
+
     private int maxPatience = 10; //Not sure if this will ever change, but the option is there
 
-    private ArrayList deck; //TODO: this script will need a reference to the deck in whatever form that may take
+    private List<int> deck = new List<int>(); //reference to a list of card id numbers that make up the deck 
+    private List<int> handBackend = new List<int>(); //purgatory state between being in the deck and being discarded.
+    private List<int> discard = new List<int>(); //list of cards that have been played and removed from the deck
+
+    private List<ConversationCard> hand = new List<ConversationCard>(); //something weird is going on with all the "hands" I'll sort it out later (probably not)
+    private Dictionary<GameObject, ConversationCard> handFrontend = new Dictionary<GameObject, ConversationCard>(); //container to hold instantiated card objects
+    private List<List<int>> filters = new List<List<int>>(); //experimental coolness
+
     private Sprite NpcSprite; //TODO: a sprite reference will need to be passed in in order to display who the player is in an encounter with
 
     public GameObject patienceBar;
     public GameObject complianceBar;
 
+    public GameObject playField;
+    private Dictionary<Transform, bool> locations = new Dictionary<Transform, bool>(); //<spawn location, location is occupied>
+
+    public GameObject redCard;
+    public GameObject blueCard;
+    public GameObject greenCard;
+    public GameObject prepCard;
+
     /**
      * initializes the ecounter. 
      * more parameters may be added as their implementaion is solidified
      */
-    public void StartEncounter(int complianceThreshold, int startingCompliance, int startingPatience)
+    public void StartEncounter(int complianceThreshold, int startingCompliance, int startingPatience, string NpcWeakness, string NpcResistance)
     {
+        if(GameState.Meta.activeEncounter.Value != null)
+        {
+            throw new Exception();
+        }
+        GameState.Meta.activeEncounter.Value = this;
+
```

Comment:
Eek! I'm not sure we should throw an exception here. I would Debug.Error() or Debug.Critical() (if it exists). I'm not sure how Unity handles unhandled exceptions, but if it is by crashing (or telling the user there was an unhandled exception) we should avoid it at all costs. If it is unavoidable, it should *definitely* have a message attached